cmd/clair-load-test/reports.go

```diff
@@ -132,20 +138,37 @@ func NewConfig(c *cli.Context) *TestConfig {
 	}
 }
 
+// calculateLayers calculates the layers number used while fetching images.
+// It returns an integer indicating amount of layers.
+func calculateLayers(ctx context.Context, layers int, validLayers []int) int {
+	if layers == (-1) {
+		index := rand.Intn(len(validLayers) - 1)
+		return validLayers[1+index]
+	} else {
+		return layers
+	}
+}
+
 // getContainersList returns list of containers from test repo used in load phase.
 // It returns a list of strings which is a list of container names.
-func getContainersList(ctx context.Context, testRepoPrefix string, hitSize int, layers int, validLayers []int) []string {
+func getContainersList(ctx context.Context, testRepoPrefix []string, hitSize, layers int, validLayers []int) []string {
 	var containers []string
-	var newLayers int
-	rand.Seed(time.Now().UnixNano())
-	for i := 1; i <= hitSize; i++ {
-		if layers == (-1) {
-			index := rand.Intn(len(validLayers) - 1)
-			newLayers = validLayers[1+index]
-		} else {
-			newLayers = layers
+	testRepos := len(testRepoPrefix)
+	imagesPerRepo := (hitSize / testRepos)
+	leftOverImages := (hitSize % testRepos)
+
+	if imagesPerRepo > 0 {
+		for _, repoPrefix := range testRepoPrefix {
+			for i := 1; i <= imagesPerRepo; i++ {
+				containers = append(containers, repoPrefix+"_layers_"+strconv.Itoa(calculateLayers(ctx, layers, validLayers))+"_tag_"+strconv.Itoa(i))
+			}
 		}
-		containers = append(containers, testRepoPrefix+"_layers_"+strconv.Itoa(newLayers)+"_tag_"+strconv.Itoa(i))
+	}
+
+	idx := 0
+	for idx < leftOverImages {
+		containers = append(containers, testRepoPrefix[idx]+"_layers_"+strconv.Itoa(calculateLayers(ctx, layers, validLayers))+"_tag_"+strconv.Itoa(imagesPerRepo+1))
```

Comment:
Sorry for the confusion, I mean encapsulating this logic:
```go
testRepoPrefix[idx]+"_layers_"+strconv.Itoa(calculateLayers(ctx, layers, validLayers))+"_tag_"+strconv.Itoa(imagesPerRepo+1)
```

Into a function like:
```go
func containerName(prefix string, nLayers, nTag int) string {
    return prefix+"_layers_"+strconv.Itoa(nLayers)+"_tag_"+strconv.Itoa(inTag)
}
```
For better self documentation and to avoid having to update in multiple places.

Reply:
Acknowledged. Changed as suggested.